### ISSUE TITLE
Segmentation fault when loading coordinators for empty-string consumer group name

### ISSUE THREAD

**Issue by hhromic** [twmb/franz-go#283]
While testing the latest version of [KMinion](https://github.com/redpanda-data/kminion) (a Kafka monitoring tool that uses `franz-go` under the hood) which recently upgraded `franz-go` from version `v1.4.1` to version `v1.10.0`, the application is now crashing with the following panic:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x7de84d]

goroutine 73 [running]:
github.com/twmb/franz-go/pkg/kgo.(*describeGroupsSharder).shard(0x800?, {0xcf0750?, 0xc000460100?}, {0xcf4828?, 0xc00001a9c0}, {0x7f2688649fff?, 0x429ea5?})
        /go/pkg/mod/github.com/twmb/franz-go@v1.10.0/pkg/kgo/client.go:2544 +0xa2d
github.com/twmb/franz-go/pkg/kgo.(*Client).handleShardedReq.func2({0x0, {0xcf4828, 0xc00001a9c0}, {0x0, 0x0}})
        /go/pkg/mod/github.com/twmb/franz-go@v1.10.0/pkg/kgo/client.go:1730 +0x151
github.com/twmb/franz-go/pkg/kgo.(*Client).handleShardedReq(0xc000278000, {0xcf0750?, 0xc000460100}, {0xcf4828?, 0xc00001a9c0})
        /go/pkg/mod/github.com/twmb/franz-go@v1.10.0/pkg/kgo/client.go:1821 +0x9f9
github.com/twmb/franz-go/pkg/kgo.(*Client).shardedRequest(0xc000278000, {0xcf07f8?, 0xc00011ac60?}, {0xcf4828?, 0xc00001a9c0})
        /go/pkg/mod/github.com/twmb/franz-go@v1.10.0/pkg/kgo/client.go:978 +0x691
github.com/twmb/franz-go/pkg/kgo.(*Client).RequestSharded(0xc000494000?, {0xcf07f8?, 0xc00011ac60?}, {0xcf4828?, 0xc00001a9c0?})
        /go/pkg/mod/github.com/twmb/franz-go@v1.10.0/pkg/kgo/client.go:909 +0x3a
github.com/cloudhut/kminion/v2/minion.(*Service).DescribeConsumerGroups(0xc000494000, {0xcf07f8, 0xc00011ac60})
        /app/minion/describe_consumer_groups.go:70 +0x15b
github.com/cloudhut/kminion/v2/prometheus.(*Exporter).collectConsumerGroups(0xc000360000, {0xcf07f8?, 0xc00011ac60?}, 0xcea010?)
        /app/prometheus/collect_consumer_groups.go:18 +0x5e
github.com/cloudhut/kminion/v2/prometheus.(*Exporter).Collect(0xc000360000, 0xc000392f60?)
        /app/prometheus/exporter.go:235 +0x205
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
        /go/pkg/mod/github.com/prometheus/client_golang@v1.14.0/prometheus/registry.go:456 +0x10d
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather
        /go/pkg/mod/github.com/prometheus/client_golang@v1.14.0/prometheus/registry.go:467 +0x55d
```

I traced back the problem to my Kafka cluster having an empty-string consumer group name. We use Confluent's Kafka Enterprise version and the Control Center component has the (bad) habit of creating these empty-string named consumer groups for itself.

When KMinion attempts to gather information about that group using `franz-go`, then it crashes as shown above. Important is to note that this was not crashing in a previous version of KMinion that used `franz-go` version `v1.4.1`, even with these empty-string name consumer groups in the cluster. `franz-go` was able to fetch metadata correctly for these groups.

I manually upgraded `franz-go` in KMinion to latest `v1.10.3` and the problem is still there.

It seems that the [`loadCoordinators()` function](https://github.com/twmb/franz-go/blob/2e4f905202516652ee08b4d8df0c499afa4a89f2/pkg/kgo/client.go#L1149) in `franz-go` is not playing well when the requested consumer group is an empty name, i.e. the string `""`, and the returned `berr.b` structure becomes `nil` for it.

Original issue opened in KMinion's repository: https://github.com/redpanda-data/kminion/issues/177

**Comment by twmb** [comment]
Good find, fix should be easy, I’ll push a patch release later today

**Comment by twmb** [comment]
Should be fixed in v1.10.4... I tested locally by describing many groups, including empty groups.
